src/components/experiences/ExperiencesForm.tsx

```diff
@@ -1,70 +1,280 @@
+                          src={platform.imageUrl}
+                          alt=""
+                        />
+                      </div>
+                      <div className="min-w-0 flex-1">
+                        <p className="truncate text-sm font-medium text-gray-900">
+                          {platform.name}
+                        </p>
+                        <p className="truncate text-sm font-medium text-gray-500">
+                          {platform.description}
+                        </p>
+                      </div>
+                    </div>
+                    <div className="flex-shrink-0"></div>
 
-        {state.isValid && (
-          <input
-            type="url"
-            name="url"
-            id="url"
-            className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
-            placeholder="Enter your experience URL"
-          />
-        )}
+                    
+                  </li>
+                ))}
+              </ul>
+            </div>
+          </fieldset>
+        </div>
 
-        {state.isValid == false && (
-          <div className="relative block w-full rounded-md shadow-sm">
-            <input
-              type="url"
-              name="url"
-              id="url"
-              className=" block w-full rounded-md border-0 py-1.5 pr-10 text-red-900 ring-1 ring-inset ring-red-300 placeholder:text-red-300 focus:ring-2 focus:ring-inset focus:ring-red-500 sm:text-sm sm:leading-6"
-              placeholder={state.url}
-              defaultValue={state.url}
-              aria-invalid="true"
-              aria-describedby="url-error"
-            />
-            <div className="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-3">
-              <ExclamationCircleIcon
-                className="h-5 w-5 text-red-500"
-                aria-hidden="true"
+        <div className="mt-6 flex">
+          <label htmlFor="url" className="sr-only">
+            {selectedPlatform === "rezdy"
+              ? `Search Term for Rezdy`
+              : `Short Name for FareHarbour`}
+          </label>
+
+          {state.mode === "success" ||
+            (state.mode === "initial" && (
+              <input
+                type="search"
+                name="url"
+                id="url"
+                className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
+                placeholder={
+                  selectedPlatform === "rezdy"
+                    ? `Search for your experiences on Rezdy`
+                    : `Short name in FareHarbour`
+                }
+              />
+            ))}
+
+          {state.mode === "error" && (
```

Comment:
I reckon this would re-render all together different component when the state changes. Would we be able to handle them with just one <input> component itself

Reply:
thanks bro.. will be address this.